spring-cloud-zuul-ratelimit-core/src/main/java/com/marcosbarbero/cloud/autoconfigure/zuul/ratelimit/config/properties/RateLimitType.java

```diff
@@ -63,14 +64,30 @@ public boolean apply(HttpServletRequest request, Route route, RateLimitUtils rat
         }
 
         @Override
-        public String key(HttpServletRequest request, Route route, RateLimitUtils rateLimitUtils) {
+        public String key(HttpServletRequest request, Route route, RateLimitUtils rateLimitUtils, String matcher) {
             return Optional.ofNullable(route).map(Route::getPath).orElse(StringUtils.EMPTY);
         }
     },
+
+    /**
+     * Rate limit policy considering the authenticated user's role.
+     */
+    ROLE {
+        @Override
+        public boolean apply(HttpServletRequest request, Route route, RateLimitUtils rateLimitUtils, String matcher) {
+            return rateLimitUtils.getUserRoles().contains(matcher.toUpperCase());
```

Comment:
Here I'm doing `matcher.toUpperCase()` because if the configuration contains it in lower case it won't match.